src/metabase/pulse.clj

```diff
@@ -1,50 +1,105 @@
+            (clj-time [coerce :as c]
+                      [core :as t]
+                      [format :as f])
+            [hiccup.core :refer [html h]]
+            [metabase.models.setting :as setting]
+            [metabase.util :as u])
+  (:import (java.awt BasicStroke
+                     Color
+                     RenderingHints)
+           java.awt.image.BufferedImage
+           java.io.ByteArrayOutputStream
+           java.util.Date
+           javax.imageio.ImageIO
+           org.apache.commons.io.IOUtils
+           (org.fit.cssbox.css CSSNorm
+                               DOMAnalyzer
+                               DOMAnalyzer$Origin)
+           (org.fit.cssbox.io DefaultDOMSource
+                              StreamDocumentSource)
+           org.fit.cssbox.layout.BrowserCanvas
+           org.fit.cssbox.misc.Base64Coder))
 
 ;; NOTE: hiccup does not escape content by default so be sure to use "h" to escape any user-controlled content :-/
 
 ;;; ## CONFIG
 
-(def ^:const card-width 400)
-(def ^:const rows-limit 10)
-(def ^:const cols-limit 3)
-(def ^:const sparkline-dot-radius 6)
-(def ^:const sparkline-thickness 3)
-(def ^:const sparkline-pad 8)
+(def ^:private ^:const card-width 400)
+(def ^:private ^:const rows-limit 10)
+(def ^:private ^:const cols-limit 3)
+(def ^:private ^:const sparkline-dot-radius 6)
+(def ^:private ^:const sparkline-thickness 3)
+(def ^:private ^:const sparkline-pad 8)
 
 ;;; ## STYLES
-
-(def ^:const color-brand  "rgb(45,134,212)")
-(def ^:const color-purple "rgb(135,93,175)")
-(def ^:const color-grey-1 "rgb(248,248,248)")
-(def ^:const color-grey-2 "rgb(189,193,191)")
-(def ^:const color-grey-3 "rgb(124,131,129)")
+(def ^:private ^:const color-brand  "rgb(45,134,212)")
+(def ^:private ^:const color-purple "rgb(135,93,175)")
+(def ^:private ^:const color-grey-1 "rgb(248,248,248)")
+(def ^:private ^:const color-grey-2 "rgb(189,193,191)")
+(def ^:private ^:const color-grey-3 "rgb(124,131,129)")
 (def ^:const color-grey-4 "rgb(57,67,64)")
 
-(def ^:const font-style    "font-family: Lato, \"Helvetica Neue\", Helvetica, Arial, sans-serif;")
-(def ^:const section-style (str font-style))
-(def ^:const header-style  (str font-style "font-size: 16px; font-weight: 700; color: " color-grey-4 "; text-decoration: none;"))
-(def ^:const scalar-style  (str font-style "font-size: 24px; font-weight: 700; color: " color-brand ";"))
-(def ^:const bar-th-style  (str font-style "font-size: 10px; font-weight: 400; color: " color-grey-4 "; border-bottom: 4px solid " color-grey-1 "; padding-top: 0px; padding-bottom: 10px;"))
-(def ^:const bar-td-style  (str font-style "font-size: 16px; font-weight: 400; text-align: left; padding-right: 1em; padding-top: 8px;"))
-(def ^:const button-style  (str font-style "display: inline-block; box-sizing: border-box; padding: 8px; color: " color-brand "; border: 1px solid " color-brand "; border-radius: 4px; text-decoration: none; "))
```

Comment:
I found the style strings a bit unwieldly and there were cases where we just merged them and got duplicate keys in the generated HTML like:

>  font-size: 10px; font-weight: 400; color: <color-grey-4>; border-bottom: 4px solid <color-grey-1>; padding-top: 0px; padding-bottom: 10px; font-size: 16px; font-weight: 400; text-align: left; padding-right: 1em; padding-top: 8px;

(when merging `bar-th-style` and `bar-td-style`)

I replaced them with a function `style` that takes one or more maps and generates the appropriate CSS